source/result.js

```diff
@@ -101,12 +98,14 @@ Object.assign(Ok.prototype, {
     return this.value;
   },
 
-  map(λ) {
```

Comment:
We need to preserve compat for this which users of a monad-like will expect there to exist, so we can add but not remove

Reply:
Additionally, I like to have something that is clear and predictable instead of something magical (which more often than not is unpredictable!)

Reply:
Well, I thought map is redundant in monads. flatMap already does that

Reply:
`map` and `flatMap` are supposed to be different in monads.

For any monad `Monad`:
`map` (or 'fmap') takes a function which has the signature: `a => b`
whereas
flatMap (or `bind`, `chain`, or `(>>=)`) takes a function with the signature `a => Monad b`